gravity-forms/gw-cache-buster.php

```diff
@@ -271,6 +272,18 @@ public function filter_resume_link( $resume_url, $form, $resume_token, $email )
 		return add_query_arg( array( 'gf_token' => $resume_token ), $referer );
 	}
 
+	public function replace_embed_url( $text, $form, $entry, $url_encode ) {
+		// Check if the text contains the {embed_url} merge tag
+		if ( strpos( $text, '{embed_url}' ) !== false ) {
+	
+			$original_url = rgget( 'form_request_origin' );
```

Comment:
@saifsultanc It seems like this will still only impact default values. 

Does it work after submission and show properly in the entry details? I think we'll need to add a hidden field to track this. The hidden field will contain what is being populated in `form_request_origin`.

Here's an example of how we do that: 

https://github.com/gravitywiz/gp-entry-blocks/blob/master/includes/blocks/class-edit-form.php#L187-L205

It's hooked here: https://github.com/gravitywiz/gp-entry-blocks/blob/master/includes/blocks/class-edit-form.php#L87

Lastly, I do think security is a concern here. Because cache buster is enabled using a shortcode, we don't really know if it's being used for a form or not. 

Because of this, I recommend encrypting the hidden field value using `GFCommon::openssl_encrypt()` and then decrypted using `GFCommon::openssl_decrypt()`. If decryption fails, we just skip modifying `source_url`. It's probably worth adding this same treatment to `form_request_origin` when we're building `$ajax_url`.

Reply:
**CONFIRMATION PAGE:** ✅
<img width="440" alt="Screenshot 2024-02-28 at 10 45 57 AM" src="https://github.com/gravitywiz/snippet-library/assets/26293394/aeea8886-f5c8-4343-a27f-1a7a76c00470">

**NOTIFICATION PAGE:** ✅
<img width="440" alt="Screenshot 2024-02-28 at 10 46 11 AM" src="https://github.com/gravitywiz/snippet-library/assets/26293394/e1897623-99cb-46a4-afa3-edfe28dcf04c">

**ENTRY VIEW:** ✅
<img width="440" alt="Screenshot 2024-02-28 at 10 46 57 AM" src="https://github.com/gravitywiz/snippet-library/assets/26293394/0421fcb9-5b43-4276-a28e-1de62c04b406">

Reply:
Loom: https://www.loom.com/share/34272213f3944d2ea67d8d321c858e53